crypto/x509/x509rset.c

```diff
@@ -19,6 +19,10 @@ int X509_REQ_set_version(X509_REQ *x, long version)
 {
     if (x == NULL)
         return 0;
+    if (version != X509_REQ_VERSION_1) {
+        ERR_raise(ERR_LIB_X509, ERR_R_PASSED_INVALID_ARGUMENT);
```

Comment:
Could you please merge this with the previous `if` as I think raising the same error for `x == NULL` would be also beneficial.